pkg/acceptance/testing.go

```diff
@@ -0,0 +1,86 @@
+package acceptance
+
+import (
+	"context"
+	"testing"
+
+	"github.com/Snowflake-Labs/terraform-provider-snowflake/pkg/provider"
+	"github.com/Snowflake-Labs/terraform-provider-snowflake/pkg/sdk"
+	"github.com/hashicorp/terraform-plugin-go/tfprotov6"
+	"github.com/hashicorp/terraform-plugin-mux/tf5to6server"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+	"github.com/hashicorp/terraform-plugin-testing/helper/resource"
+)
+
+func TestMain(m *testing.M) {
+	resource.TestMain(m)
+}
+
+var (
+	TestDatabaseName  = "terraform_test_database"
+	TestSchemaName    = "terraform_test_schema"
+	TestWarehouseName = "terraform_test_warehouse"
+)
+
+var TestAccProvider *schema.Provider
+
+var TestAccProtoV6ProviderFactories = map[string]func() (tfprotov6.ProviderServer, error){
+	"snowflake": func() (tfprotov6.ProviderServer, error) {
+		return tf5to6server.UpgradeServer(
+			context.Background(),
+			TestAccProvider.GRPCProvider,
+		)
+	},
+}
+
+func init() {
+	TestAccProvider = provider.Provider()
+}
+
+func TestAccProviders() map[string]*schema.Provider {
+	p := provider.Provider()
+	return map[string]*schema.Provider{
+		"snowflake": p,
+	}
+}
+
+func TestAccPreCheck(t *testing.T) {
+	client, err := sdk.NewDefaultClient()
+	if err != nil {
+		t.Fatal(err)
+	}
+	ctx := context.Background()
+
+	// create test database
```

Comment:
i simplified it, so no longer needing to do the read before the create. since its already only one function, I don't think breaking it into separate private functions makes sense